core/src/util.jl

```diff
@@ -887,3 +896,66 @@ end
 (A::AbstractInterpolation)(t::GradientTracer) = t
 reduction_factor(x::GradientTracer, threshold::Real) = x
 relaxed_root(x::GradientTracer, threshold::Real) = x
+get_area_and_level(basin::Basin, state_idx::Int, storage::GradientTracer) = storage, storage
+adapt_negative_storage_du!(du, u::ComponentVector{<:GradientTracer}) = nothing
+
+@kwdef struct MonitoredBackTracking{B, V}
+    linesearch::B = BackTracking()
+    dz_tmp::V = []
+    z_tmp::V = []
+end
+
+function resid(z, integrator, nlsolver, f)
```

Comment:
Needs a docstring.

Reply:
Or a rename, now I think of it as res_id, not residual. 🤷🏻‍♂️